compiler/AST/WhileDoStmt.cpp

```diff
@@ -31,6 +31,13 @@
 *                                                                           *
 ************************************* | ************************************/
 
+static CallExpr* isPrimIfVar(Expr* cond) {
+  if (CallExpr* call = toCallExpr(cond))
+    if (call->isPrimitive(PRIM_IF_VAR))
```

Comment:
It'd be good to rename the primitive to something that encapsulates all the cases it is used for, but I am fine with not doing that today so that this PR can get merged in time.